helper/Data.php

```diff
@@ -336,4 +337,25 @@ public function getTemplateFromModulePath($templatePath)
 
         return 'module:adyenofficial/' . ltrim($templatePath, '/');
     }
+
+    /**
+     * Execute basic SELECT statement. If table does not exist, catch exception and return false
+     *
+     * @param $tableName
+     * @return bool
+     *
+     */
+    public function tableExists($tableName)
+    {
+        $db = Db::getInstance();
+        $sql = 'SELECT * FROM ' . _DB_PREFIX_ . $tableName;
```

Comment:
Maybe this could use a LIMIT 1 so that there's no unnecessary stress to fetch all rows. Or a different command like SHOW TABLES

Reply:
Function called eventually, automatically adds Limit 1:
`Db::getRow`:
```
    /**
     * Returns an associative array containing the first row of the query
     * This function automatically adds "LIMIT 1" to the query.
     *
     * @param string|DbQuery $sql the select query (without "LIMIT 1")
     * @param bool $use_cache Find it in cache first
     *
     * @return array|bool|object|null
     */
    public function getRow($sql, $use_cache = true)
```